### PR TITLE
Users can now send images in chat prototype

### PR DESCRIPTION
**Pull request by Scar26**
#330 

Now:
imgup

### CODE REVIEW

File: activities/ChatPrototype.activity/js/activity.js [hunk 1]
```diff
@@ -124,6 +144,18 @@ define(["sugar-web/activity/activity","webL10n","sugar-web/graphics/palette","su
 				return false;
             }
         };
+
+        imageUpload.onchange = function(){
+        	var image = imageUpload.files[0];
+			var reader = new FileReader();
```
Comment (llaske) [inline activities/ChatPrototype.activity/js/activity.js]: For compatibility, images should be taken only from the Journal. See how to include the Journal chooser dialog box here: https://github.com/llaske/sugarizer/blob/dev/docs/tutorial_step7.md

**Comment by Scar26** [comment]
Journal integrated.
Now:
journal

Defined a separate function for sending data through the presenceObject making it easier to accommodate future updates.

Also noticed a slight discrepancy in the code given in https://github.com/llaske/sugarizer/blob/dev/docs/tutorial_step7.md

**Comment by Scar26** [comment]
@llaske Are there any other improvements I could make in this PR?

**Comment by Scar26** [comment]
@llaske  Are there any changes I should make with respect to this PR?

**Comment by Scar26** [comment]
@llaske Are there any changes I should make with respect to this PR?

**Comment by Scar26** [comment]
@llaske  I have made the requested changed. Can you please review this PR?

**Comment by llaske** [comment]
Could you solve conflict on last dev branch?

**Comment by Scar26** [comment]
done @llaske

**Comment by llaske** [comment]
Nice work.
Few remarks:
- The icon quality is bad, should be nice to use the "standard" Sugar icon to include an image
- It will be better to put the icon into the toolbar
- The user should not be able to integrate an image before connection
- You broke compatibility with a client that do not include this feature. So an "old" version of the app will not be able to communicate with this new version
- Please fix conflict

**Comment by Scar26** [comment]
Now : 
toolbar

-  Icon changed. This is the icon used in record activity as well as QRcode activity.
-  Image Upload icon moved to toolbar
-  The icon is online visible when the user connected, therefore they cannot integrate an image before connection
-  Conflict fixed

An old version of the app will still be able to exchange text messages with a newer version since I'm only sending a new member "type" in the message object. It just won't be used anywhere. 

In the case that a user on the newer version sends an image, the old version will display the base64 code of the image. Currently, I see no way of dealing with this bug. Even with changes to the Sugarizer server, note that the userInfo object that is sent to the server does not contain the application version so the server won't be able to decipher what to send to a socket. so this is just a glitch that's gonna persist unless everyone in the neighborhood updates to the new version.

**Comment by llaske** [comment]
Nice. The image is good now.
There is still a conflict: If an old client try to send a message to the new client, the new client don't display the old client message (just ignore it). BTW old client is able to display new client message (expect images but it's okay).

**Comment by Scar26** [comment]
Hm...I'll look into it

**Comment by Scar26** [comment]
Done @llaske 

Now older versions can interact with the new one

**Comment by Scar26** [comment]
@llaske I have made the requested changes, can you please Review this PR?

File: activities/ChatPrototype.activity/css/activity.css [hunk 2]
```diff
@@ -136,17 +142,21 @@ textarea {
   background-repeat: no-repeat;
   margin-top: 8px;
   margin-left: 10px;
+  font-size: 0;
```
Comment (llaske) [inline activities/ChatPrototype.activity/css/activity.css]: Should be removed to follow https://github.com/llaske/sugarizer/pull/323

File: activities/ChatPrototype.activity/css/activity.css [hunk 3]
```diff
@@ -136,17 +142,21 @@ textarea {
...
 }
 
 #sad-button {
   position: absolute;
   margin-top:8px;
   margin-left: 55px;
+  font-size: 0;
```
Comment (llaske) [inline activities/ChatPrototype.activity/css/activity.css]: Should be removed to follow https://github.com/llaske/sugarizer/pull/323

File: activities/ChatPrototype.activity/icons/insert-picture.svg [hunk 4]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (llaske) [inline activities/ChatPrototype.activity/icons/insert-picture.svg]: Do you still need this file?

File: main.js [hunk 5]
```diff
@@ -119,6 +119,7 @@ function createWindow () {
 		mainWindow.maximize();
 	}
 
+
```
Comment (llaske) [inline main.js]: Do not upgrade this file

**Comment by Scar26** [comment]
Done @llaske

**Comment by llaske** [comment]
Ok. Nice.

**Pull request merged**